rfcs/convergence/authoring-stories.md

```diff
@@ -267,13 +270,40 @@ TBA
 
 TBA
 
-### 7. how structure stories in the storybook nav tree
+### 7. dissecting big story files into smaller ones
 
-TBA
+Components with complex API require many long stories, both for documentation and for e2e testing. This creates long files with hundres of lines, which are hard to maintain. How can these files be dissected into smaller ones?
+
+**Proposal:**
 
-### 8. dissecting big story files into smaller ones
+1. Every component can have at most one `.stories.tsx` file. This file has a default export with Storybook configuration for the docs page.
+2. If putting all stories into one file would make it too long, individual stories might be put into `.story.tsx` files as a named export, and then re-exported from `.stories.tsx` file like this: `export * from ‘./IndividualStoryFile.story’;`
+3. If individual story files are employed, `.stories.tsx` file must not contain any stories.
 
-TBA
+### 8. location and naming convention
+
+1. Story files should be in the same folder as the component which they are targeting.
```

Comment:
Let's go into detail about `the component which they are targeting`

i.e. if I wanted to show off `MenuItemCheckbox` which can't function without `Menu`, should that go in the `Menu` folder or the `MenuItemCheckbox` folder ?

Reply:
You want to show off `MenuItemCheckbox`, so it should go into `MenuItemCheckbox` folder. 

Is this better?

```suggestion
1. Story files should be in the same folder as the component which they are meant to show off. This will usually be the component which is the most unique to the story.
```

Reply:
maybe it would make sense to set following rule - only package public API  should have 1:1 stories that are rendered in SB tree view, WDYT ?

Reply:
To clarify - are you suggesting that every publicly available component must have _exactly one_ .stories.tsx file? I think _at most one_ is a better rule, as for some subcomponents, separate page is not necessary. Examples would be `AccordionItem`, `MenuGroup`, ...

Reply:
yeah what you said. there is another rabbit hole though :D , when you mentioned subcomponents - we probably should use generated `controls` for those as well, which will complicate things a lot ... but we can address that in another PR